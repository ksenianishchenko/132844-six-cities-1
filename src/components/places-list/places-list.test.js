import React from "react";
import renderer from "react-test-renderer";
import PlacesList from "./places-list.jsx";
import {createStore} from 'redux';
import combineReducers from "../../reducers/index.js";
import {Provider} from 'react-redux';
import {BrowserRouter} from "react-router-dom";

const places = [
  {
    "id": 1,
    "city": {
      "name": `Amsterdam`,
      "location": {
        "latitude": 52.370216,
        "longitude": 4.895168,
        "zoom": 10
      }
    },
    "preview_image": `img/1.png`,
    "images": [`img/1.png`, `img/2.png`],
    "title": `Beautiful & luxurious studio at great location`,
    "is_favorite": false,
    "is_premium": false,
    "rating": 4.8,
    "type": `apartment`,
    "bedrooms": 3,
    "max_adults": 4,
    "price": 120,
    "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    "host": {
      "id": 3,
      "is_pro": true,
      "name": `Angelina`,
      "avatar_url": `img/1.png`
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "location": {
      "latitude": 52.35514938496378,
      "longitude": 4.673877537499948,
      "zoom": 8,
    }
  },
];

const city = {"name": `Dusseldorf`,
  "location": {"latitude": 51.225402, "longitude": 6.776314, "zoom": 13},
};

it(`List of places renders correctly`, () => {
  const store = createStore(combineReducers);
  const tree = renderer
  .create(<Provider store={store}>
    <BrowserRouter>
      <PlacesList
        places = {places}
        activeCity = {city}
        cities = {[city]}
        activeElement = {`Popular`}
      />
    </BrowserRouter>
  </Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
