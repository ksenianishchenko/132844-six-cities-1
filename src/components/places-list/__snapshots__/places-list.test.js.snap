// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List of places renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PlacesList
    activeCity={
      Object {
        "location": Object {
          "latitude": 51.225402,
          "longitude": 6.776314,
          "zoom": 13,
        },
        "name": "Dusseldorf",
      }
    }
    activeElement="Popular"
    cities={
      Array [
        Object {
          "location": Object {
            "latitude": 51.225402,
            "longitude": 6.776314,
            "zoom": 13,
          },
          "name": "Dusseldorf",
        },
      ]
    }
    places={
      Array [
        Object {
          "bedrooms": 3,
          "city": Object {
            "location": Object {
              "latitude": 52.370216,
              "longitude": 4.895168,
              "zoom": 10,
            },
            "name": "Amsterdam",
          },
          "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
          "goods": Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
            "Coffee machine",
            "Dishwasher",
          ],
          "host": Object {
            "avatar_url": "img/1.png",
            "id": 3,
            "is_pro": true,
            "name": "Angelina",
          },
          "id": 1,
          "images": Array [
            "img/1.png",
            "img/2.png",
          ],
          "is_favorite": false,
          "is_premium": false,
          "location": Object {
            "latitude": 52.35514938496378,
            "longitude": 4.673877537499948,
            "zoom": 8,
          },
          "max_adults": 4,
          "preview_image": "img/1.png",
          "price": 120,
          "rating": 4.8,
          "title": "Beautiful & luxurious studio at great location",
          "type": "apartment",
        },
      ]
    }
  />
</Router>
`;
